### PR TITLE
Update AdaptiveSkillDialog to use ActivityProcessed

### LINKED ISSUE #3772: Update AdaptiveSkillDialog to use ActivityProcessed
## Version
What package version of the SDK are you using.

## Describe the bug
In the AdaptiveSkillDialog, since the activity is sent explicitly, I don’t understand how activityProcessed is used. 
Could we just remove that from the code and schema? Otherwise we need to expose it in Composer, and we can’t explain to user how this value take effect. 

@gabog, @tomml, @carlosscastro, @vishwacsena  

## Expected behavior
Give a clear and concise description of what you expected to happen.

## Additional context

[bug]

### PR DESCRIPTION
**Pull request by gabog**
Fixes #3772

Updated AdaptiveSkillDialog to use the ActivityProcessed property and updated documentation.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs [hunk 1]
```diff
@@ -32,18 +32,15 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
-                {
-                    return obj.ToObject<ActivityTemplate>();
-                }
-
-                var activity = obj.ToObject<Activity>();
-                return new StaticActivityTemplate((Activity)activity);
+                return obj.ToObject<ActivityTemplate>();
             }
+
+            return obj.ToObject<StaticActivityTemplate>();
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs:43]: > return obj.ToObject<StaticActivityTemplate>(); [](start = 11, length = 47)

This is wrong. The serialized object is not a staticActivityTemplate, it's a StaticActivityTemplate wrapped around a serialized Activity.  I'm super suprised unit tests worked with this change.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        [TestMethod]
+        public void ReadsStaticActivityTemplateActivityTemplateJson()
+        {
+            var jsonText = @"{'activity': {
+            '$kind': 'Microsoft.StaticActivityTemplate',
+            'activity': {
+                'type': 'event',
+                'name': 'BookFlight'
+            }
+            }}";
```
Comment (tomlm) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs:39]: This is wrong.  What we are modeling is that

prompt => "Hello world"  
prompt => {  ... ACTIVITY .... }
prompt => {  "$kind":"Sometype which implements ITemplate<Activity>" ....

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs [hunk 3]
```diff
@@ -38,16 +38,6 @@ public AdaptiveSkillDialog([CallerFilePath] string callerPath = "", [CallerLineN
...
-        /// Gets or sets a value indicating whether to have the new dialog should process the activity.
-        /// </summary>
-        /// <value>
-        /// The default for this will be true, which means the new dialog should not look the activity. You can set this to false to dispatch the activity to the new dialog.
-        /// </value>
-        [DefaultValue(true)]
-        [JsonProperty("activityProcessed")]
-        public BoolExpression ActivityProcessed { get; set; } = true;
-
         /// <summary>
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs:41]: You shouldn't remove this.  If you do this you have no way of controlling whether the skill should be doing recognition on the activity or not.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs (hunk 1 above)
Comment (gabog) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs:43]: I had to do this to get tests\Microsoft.Bot.Builder.TestBot.Json\Samples\80 - AdaptiveSkillDialog\AdaptiveSkillDialog.main.dialog in TestJson to work, it was working until a while ago but not the project wasn't able to get the Activity property in AdaptiveSkillDialog.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs (hunk 2 above)
Comment (gabog) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs:39]: See tests\Microsoft.Bot.Builder.TestBot.Json\Samples\80 - AdaptiveSkillDialog\AdaptiveSkillDialog.main.dialog this is what the test is based on.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs (hunk 3 above)
Comment (gabog) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs:41]: @luhan2017 requested we remove this and @vishwacsena also agreed (see issue linked to the PR), we don't seem to be understand how is being used by composer or skilldialog.

Comment (luhan2017) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs:41]: I don't see how this activityProcessed is passed to and consumed by the skill bot. Do we pass the turn memory of the caller bot to the skill bot? 
In the SkillDialog, the activity is not from the last turn, it just comes from the property of the SkillDialog, so I feel that the meaning of the activityProcessed in BeginDialog is not same as the meaning of whether we need to go through intent recognition in skill, should we make another setting specially for this?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs (hunk 1 above)
Comment (gabog) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/ActivityTemplateConverter.cs]: Reverted to your change from master that addresses the issue I was having loading the activity template.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs (hunk 2 above)
Comment (gabog) [reply tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs]: Updated AdaptiveSkillDialog.main.dialog to use the simpler notation and manually tested it verify it works after merging latest from master with the new serialization and schema updates.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs (hunk 3 above)
Comment (gabog) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs]: I added code in AdaptiveSkillDialog to use the ActivityProcessed property and updated the code documentation to describe its behavior.

Comment (gabog) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Skills/AdaptiveSkillDialog.cs]: Hi @luhan2017, ping me over Teams if you want to chat about this one a bit more.

**Pull request merged**